src/utils.ts

```diff
@@ -293,14 +294,51 @@ export function getEnsTextRecord(ens: string, record: string, network = '1') {
   return call(provider, ENS_RESOLVER_ABI, [address, 'text', [ensHash, record]]);
 }
 
-export async function getSpaceUri(id, network = '1') {
+export async function getSpaceUri(
+  id: string,
+  network = '1'
+): Promise<string | null> {
   try {
     return await getEnsTextRecord(id, 'snapshot', network);
   } catch (e) {
-    return false;
+    console.log(e);
+    return null;
   }
 }
 
+export async function getEnsOwner(
+  ens: string,
+  network = '1'
+): Promise<string | null> {
+  const registryAddress = '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e';
+  const provider = getProvider(network);
+  const ensRegistry = new Contract(
+    registryAddress,
+    ['function owner(bytes32) view returns (address)'],
+    provider
+  );
+  const ensHash = hash(normalize(ens));
+  return await ensRegistry.owner(ensHash);
+}
+
+export async function getSpaceController(
+  id: string,
+  network = '1'
+): Promise<string | null> {
+  const spaceUri = await getSpaceUri(id, network);
+  if (spaceUri) {
+    let isUriAddress = isAddress(spaceUri);
+    if (isUriAddress) return spaceUri;
+
+    const uriParts = spaceUri.split('/');
+    const position = uriParts.includes('testnet') ? 5 : 4;
```

Comment:
I think better to check the IPNS format as before right? not required?

Reply:
Maybe not required. Let's merge this then? works fine to me, not sure if i miss some case 😅

Reply:
Yes can merge, should cover all cases as this is how we have been doing it on the frontend too